### PR TITLE
Fixing netstandard build of System.Collection.Imutable.Tests project

### PR DESCRIPTION
**Pull request by joperezr**
In order to be able to produce a .NET Standard test suite (issue #30729), first we have to fix all test projects that currently have a netstandard configuration but it doesn't build. The reason why we never realized that they don't build succesfully, is because these projects have a more specific configuration for all of our verticals, so the netstandard config never gets chosen. For this reason, most of these projects have been broken for ever and we never realized it.

These are the required changes to have the netstandard build of System.Collections.Immutable building succesfully and running all tests.

cc: @danmosemsft

### CODE REVIEW

**Comment by joperezr** [comment]
cc: @safern

File: src/System.Collections.Immutable/tests/ImmutableDictionaryTestBase.cs [hunk 1]
```diff
@@ -347,93 +243,6 @@ public void TryGetKey()
...
-            Assert.False(map.ContainsKey(key));
-            Assert.True(addedMap.ContainsKey(key));
-            AssertAreSame(value, addedMap.GetValueOrDefault(key));
-
-            this.VerifyAvlTreeState(addedMap);
-
-            return addedMap;
-        }
-
-        protected void AddAscendingTestHelper(IImmutableDictionary<int, GenericParameterHelper> map)
```
Comment (safern) [inline src/System.Collections.Immutable/tests/ImmutableDictionaryTestBase.cs]: Why would we need to move this to a nonnetstandard configuration without changing ImmutablesTestBase?

**Comment by joperezr** [comment]
> How are we going to ensure that this is not broken again when people add more tests? I don't see you changing the configurations. Are we planning on adding a netstandard tests vertical?

Yup, once I'm done with the rest of failing projects (around 3-4 more) I will add a vertical that will produce this Netstandard suite and will ensure that these configurations won't get broken again.

File: src/System.Collections.Immutable/tests/ImmutableDictionaryTestBase.cs (hunk 1 above)
Comment (joperezr) [reply src/System.Collections.Immutable/tests/ImmutableDictionaryTestBase.cs]: ImmutableTestBase has about 16 derived types, some of which required no changes to its properties and we do in fact use them in some tests. For some other derived types, the implementation of those properties uses types not available in netstandard, which is why we only split those derived types.

**Comment by joperezr** [comment]
@dotnet-bot test Linux-musl x64 Debug Build please

**Pull request merged**